### PR TITLE
Add script to convert old ggml files to newer version

### PR DESCRIPTION
**Pull request by thement**
Followup to: https://github.com/ggerganov/llama.cpp/pull/526

### CODE REVIEW

File: convert-unversioned-ggml-to-ggml.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    struct_fmt = "i" * (3 + len(HPARAMS))
+    struct_size = struct.calcsize(struct_fmt)
+    buf = f_in.read(struct_size)
+    return struct.unpack(struct_fmt, buf)
+
+def write_header(f_out, header):
+    (magic, vocab_size, dim, multiple_of, n_heads, n_layers, rot, ftype) = header
+
+    if magic != 0x67676d6c:
+        raise Exception('Invalid file magic. Must be an old style ggml file.')
```
Comment (anzz1) [inline convert-unversioned-ggml-to-ggml.py:29]: wait im confused, isn't it supposed to be old style ggml file for this?

File: llama.cpp [hunk 2]
```diff
@@ -320,7 +320,7 @@ static bool llama_model_load(
         uint32_t magic;
         fin.read((char *) &magic, sizeof(magic));
         if (magic == LLAMA_FILE_MAGIC_UNVERSIONED) {
-            fprintf(stderr, "%s: invalid model file '%s' (too old, regenerate your model files!)\n",
+            fprintf(stderr, "%s: invalid model file '%s' (too old, regenerate your model files or convert them with convert-unversioned-ggml-to-ggml.py!)\n",
```
Comment (anzz1) [inline llama.cpp:323]: this shouldn't be changed, as the conversion script is unsupported.

**Review by anzz1 — CHANGES_REQUESTED** [review]

**Comment by anzz1** [comment]
I think it would be good to print a disclaimer that the script is unsupported and results are not guaranteed, and to not post any issues regarding models generated with this.

File: convert-unversioned-ggml-to-ggml.py (hunk 1 above)
Comment (thement) [reply convert-unversioned-ggml-to-ggml.py:29]: It expects the unversioned GGML model and produces the versioned one.
`#define LLAMA_FILE_MAGIC_UNVERSIONED 0x67676d6c // pre-versioned files`

**Comment by thement** [comment]
> I think it would be good to have the script print a disclaimer that the script is unsupported and results are not guaranteed, and to not post any issues regarding models generated with this.

The model isn't all that different. The only missing thing is score in vocabulary and I could theoretically fill that in from `tokenizer.model`.

**Review by ggerganov — APPROVED** [review]
Let's add this for now, but it will eventually be removed once everyone updates their models

**Pull request merged**